packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx

```diff
@@ -207,52 +235,77 @@ export function ThirdwebWalletProvider(
     }
 
     autoConnectTriggered.current = true;
-    (async () => {
-      const lastConnectedWalletInfo = await coordinatorStorage.getItem(
-        LAST_CONNECTED_WALLET_STORAGE_KEY,
-      );
 
-      if (!lastConnectedWalletInfo) {
+    async function autoconnect() {
+      const walletInfo = await getLastConnectedWalletInfo();
+
+      if (!walletInfo) {
         setConnectionStatus("disconnected");
         return;
       }
 
-      let parsedParams: ConnectParams<Record<string, any>> | undefined;
-      let lastConnectedWalletId: string | undefined;
-
-      try {
-        const parsedWallet = JSON.parse(lastConnectedWalletInfo as string);
-        parsedParams = parsedWallet.connectParams;
-        lastConnectedWalletId = parsedWallet.walletId;
-      } catch (error) {
-        console.error(`Error parsing the last connected wallet: ${error}`);
-      }
+      const walletObj = props.supportedWallets.find(
+        (W) => W.id === walletInfo.walletId,
+      );
 
-      if (!lastConnectedWalletId) {
+      if (!walletObj) {
+        // last connected wallet is no longer present in the supported wallets
         setConnectionStatus("disconnected");
         return;
       }
 
-      let lastConnectedWallet = props.supportedWallets.find(
-        (W) => W.id === lastConnectedWalletId,
-      );
+      const personalWalletInfo = walletInfo.connectParams?.personalWallet;
 
-      if (!lastConnectedWallet) {
-        setConnectionStatus("disconnected");
-        return;
+      if (personalWalletInfo) {
+        const personalWallets =
+          walletObj?.config?.personalWallets || props.supportedWallets;
```

Comment:
Should we just look in "personalWallets" instead of both? It may look like a bug if a user is able to auto-connect with a wallet not specified in "personalWallets"

Reply:
Yeah, makes sense. Adding defaults for personalWallets